apps/web/pages/v2/settings/security/sso.tsx

```diff
@@ -0,0 +1,39 @@
+import { useRouter } from "next/router";
+
+import SAMLConfiguration from "@calcom/features/ee/sso/components/SAMLConfiguration";
+import { HOSTED_CAL_FEATURES } from "@calcom/lib/constants";
+import { trpc } from "@calcom/trpc/react";
+import Meta from "@calcom/ui/v2/core/Meta";
+import SkeletonLoader from "@calcom/ui/v2/core/apps/SkeletonLoader";
+import { getLayout } from "@calcom/ui/v2/core/layouts/SettingsLayout";
+
+const SAMLSSO = () => {
+  const router = useRouter();
+
+  if (HOSTED_CAL_FEATURES) {
+    router.replace("/404");
+  }
+
+  const teamsView = false;
+
+  const { data: saml, isLoading } = trpc.useQuery(["viewer.saml.access", { teamsView }], {
```

Comment:
The access endpoint does not throw when access is denied, instead the response is success:

<img width="332" alt="image" src="https://user-images.githubusercontent.com/1046695/194729221-b11ed77e-bcf9-4518-95b5-50274117bb2b.png">

Ideally the access menu item is hidden when the server does not support it, if the server does support it but the user can set something up that allows access, the menu should be shown with the steps needed taking to obtain access.

Currently if access is false an empty saml page is shown:

<img width="1108" alt="image" src="https://user-images.githubusercontent.com/1046695/194729339-7fb53364-c844-4b63-b52f-445bec7b0784.png">

Reply:
@emrysal Thanks for your feedback. I'll make the changes based on your feedback.

There are no significant changes compared to our initial implementation. Just some refactoring and UI changes.

The SAML SSO was there in 2 places in the original implementation. So kept it as it was. The Pro plan check was there already. I assumed only PRO users could access the SAML SSO. Am I right?

I'll discuss the above points with @deepakprabhakara and will make to a plan.

Reply:
@emrysal We already make the right check and show/hide the SSO menu item. We'll handle the access denied case and show an error message to contact the admin (which is the only logical step possible in this case, let us know if you'd like to handle it in another way).

We'll make changes to hide the SSO menu item in Teams for self-hosted and vice versa for hosted version (hide SSO menu item in Settings -> Security) since there needs to only be one place for this in both scenarios.

Please let us know if the PRO user check for access regarding SSO has changed and we'll make the right check.

Reply:
@PeerRich Could you update with the specifics on the PRO plan and how it impacts SAML post 2.0?